### DISCUSSION TITLE
enh: support external image urls in chat completion requests

### DISCUSSION THREAD

**Discussion by quanliew28** [open-webui/open-webui#19357] [Ideas]
## Installation Method
Docker

## Environment

- **Open WebUI Version:** [e.g., v0.3.35]
- **Ollama (if applicable):** [e.g., v0.4.1]

- **Operating System:** Windows 11

**Confirmation:**

- [👍 ] I have read and followed all the instructions provided in the README.md.
- [👍 ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [👍 ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
With Open AI API, it should have pass the image url to openweb ui

## Actual Behavior:

When passing Image URL to open webUI API server, it shows:
```
 Internal Server Errorcurl: (3) URL rejected: Malformed input to a URL function
curl: (6) Could not resolve host: xn--8vh0o
```

## Description

**Bug Summary:**

When passing Image URL to open webUI API server, it shows:
```
 Internal Server Errorcurl: (3) URL rejected: Malformed input to a URL function
curl: (6) Could not resolve host: xn--8vh0o
```

## Reproduction Details

**Steps to Reproduce:**
Use bash and type in this
Remember to replace API KEY and PUBLIC IMAGES
```
 curl -X POST http://localhost:3000/api/chat/completions \                                                                                                                     ─╯
-H "Authorization: Bearer [REPLACE_WITH_YOUR_OWN_API_KEY_FROM_WEBUI]" \
-H "Content-Type:application/json" \
-d '{
      "model": "qwen2.5:7b",
      "messages": [
        {
          "role": "user",
          "content": [
          {
            "type": "text",
            "text": "What'\''s in this image?"
          },
          {
            "type": "image_url",
            "image_url": {
              "url": "[REPLACE_WITH_YOUR_OWN_PUBLIC_ONLINEHOSTED_IMAGES].jpg",
            }
          }
        ],
        }
      ]
    }'
```

## Logs and Screenshots

**Screenshots/Screen Recordings (if applicable):**
![Screenshot 2024-11-10 015619](https://github.com/user-attachments/assets/94927eb7-b1c1-4759-9bd3-af3e909740a1)

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

**Comment by maxi1134** [comment]
I would also love being able to pipe images in any other fashion than Base64!

**Comment by gvkaura** [comment]
> I would also love being able to pipe images in any other fashion than Base64!

@maxi1134 i was also looking for such option. Meanwhile I wanted to understand if you are able to show images via base64 format in the chat response for non image gen models?
